tests/test_modeling.py

```diff
@@ -0,0 +1,913 @@
+    NeuronModelForMultipleChoice,
+    NeuronModelForQuestionAnswering,
+    NeuronModelForSequenceClassification,
+    NeuronModelForTokenClassification,
+)
+from optimum.neuron.utils import NEURON_FILE_NAME
+from optimum.neuron.utils.testing_utils import is_inferentia_test
+from optimum.utils import (
+    CONFIG_NAME,
+    logging,
+)
+from optimum.utils.testing_utils import require_hf_token
+
+from .exporters.exporters_utils import EXPORT_MODELS_TINY as MODEL_NAMES
+from .exporters.exporters_utils import SEED
+
+
+logger = logging.get_logger()
+
+
+class NeuronModelTestMixin(unittest.TestCase):
+    ARCH_MODEL_MAP = {}
+    STATIC_INPUTS_SHAPES = {"batch_size": 1, "sequence_length": 32}
+
+    @classmethod
+    def setUpClass(cls):
+        cls.neuron_model_dirs = {}
+
+    def _setup(self, model_args: Dict):
+        """
+        Exports the PyTorch models to Neuron models ahead of time to avoid multiple exports during the tests.
+        We don't use unittest setUpClass, in order to still be able to run individual tests.
+        """
+        model_arch = model_args["model_arch"]
+        model_arch_and_params = model_args["test_name"]
+        dynamic_batch_size = getattr(model_args, "dynamic_batch_size", False)
+
+        if model_arch_and_params not in self.neuron_model_dirs:
+            # model_args will contain kwargs to pass to NeuronModel.from_pretrained()
+            model_args.pop("test_name")
+            model_args.pop("model_arch")
+            model_args.pop("dynamic_batch_size", None)
+
+            model_id = (
+                self.ARCH_MODEL_MAP[model_arch] if model_arch in self.ARCH_MODEL_MAP else MODEL_NAMES[model_arch]
+            )
+            set_seed(SEED)
+            neuron_model = self.NEURON_MODEL_CLASS.from_pretrained(
+                model_id, **model_args, export=True, dynamic_batch_size=dynamic_batch_size, **self.STATIC_INPUTS_SHAPES
+            )
+
+            model_dir = tempfile.mkdtemp(prefix=f"{model_arch_and_params}_{self.TASK}_")
+            neuron_model.save_pretrained(model_dir)
+            self.neuron_model_dirs[model_arch_and_params] = model_dir
+
+    @classmethod
+    def tearDownClass(cls):
+        for _, dir_path in cls.neuron_model_dirs.items():
+            shutil.rmtree(dir_path)
+
+
+@is_inferentia_test
+class NeuronModelIntegrationTest(unittest.TestCase):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.LOCAL_MODEL_PATH = "assets/neuron"
+        self.NEURON_MODEL_ID = "optimum/tiny-random-BertModel-neuron"
+        self.TINY_SUBFOLDER_MODEL_ID = "fxmarty/tiny-bert-sst2-distilled-subfolder"
+        self.FAIL_NEURON_MODEL_ID = "sshleifer/tiny-distilbert-base-cased-distilled-squad"
+        self.PRIVATE_NEURON_MODEL_ID = "Jingya/tiny-random-BertModel-neuron-private"
+        self.TINY_MODEL_REMOTE = "Jingya/tiny-random-bert-remote-code"
+        self.INPUTS_SHAPES = {"batch_size": 3, "sequence_length": 64}
```

Comment:
Why not make this class attributes directly?

Reply:
Make sense, will do!